### PR TITLE
Refactor to use new assessment renderer API.

### PR DESCRIPTION
**Pull request by rtibbles**
Also clean up interactions with perseus itemRenderer.

### CODE REVIEW

**Comment by indirectlylit** [comment]
should we start using master/release/develop branches & tags for this repo?

**Comment by rtibbles** [comment]
I think having release branches in case of a need for patch bug fixes would
make sense, but except for consistency with Kolibri, I don't see a strong
need for a develop branch (note: this means that I am quite happy to do it
just for consistency).

**Comment by indirectlylit** [comment]
okay I'm not in a rush

**Comment by rtibbles** [comment]
Already did it :)

File: kolibri_exercise_perseus_plugin/assets/src/vue/index.vue [hunk 1]
```diff
@@ -1,151 +1,327 @@
...
+              Object.prototype.hasOwnProperty.call(obj.answerArea, key) &&
+              typeof obj.answerArea[key] !== 'boolean'), true) &&
+            // Check that the 'hints' property is an Array.
+          Array.isArray(obj.hints) &&
+          obj.hints.reduce(
+            // Check that each hint in the hints array is an object (and not null)
+            (prev, item) => item && typeof item === 'object', true) &&
+          // Check that the question property is an object (and not null)
+          obj.question && typeof obj.question === 'object';
+          /* eslint-enable no-mixed-operators */
```
Comment (indirectlylit) [inline kolibri_exercise_perseus_plugin/assets/src/vue/index.vue:194]: There's some very odd whitespace stuff happening in this function. Recommend that all parens & brackets open and close at the same level of indentation.

File: kolibri_exercise_perseus_plugin/assets/src/vue/index.vue [hunk 2]
```diff
@@ -1,151 +1,327 @@
...
+          'calculator',
+          'chi2Table',
+          'periodicTable',
+          'tTable',
+          'zTable',
+        ].reduce(
+            /* eslint-disable no-mixed-operators */
+            // Loop through all of the above properties and ensure that if the 'answerArea'
+            // property of the item has them, then their values are set to Booleans.
+            (prev, key) => !(!prev ||
```
Comment (indirectlylit) [inline kolibri_exercise_perseus_plugin/assets/src/vue/index.vue:184]: for this and the other fat arrow reducer, perhaps using named function syntax would help "self-document" what the reducer does?

File: kolibri_exercise_perseus_plugin/assets/src/vue/index.vue [hunk 3]
```diff
@@ -1,151 +1,327 @@
...
     }),
-    computed: {
-      recentAttempts() {
-        if (!this.pastattempts) {
-          return [];
+
+    methods: {
+      validateItemData(obj) {
+        return [
+          // A somewhat protracted validator to ensure that our item data conforms
```
Comment (indirectlylit) [inline kolibri_exercise_perseus_plugin/assets/src/vue/index.vue:172]: could always move to an external js file if it's taking up too much room here

File: kolibri_exercise_perseus_plugin/assets/src/vue/index.vue [hunk 4]
```diff
@@ -1,151 +1,327 @@
...
-        // map the list of attempt objects to simple strings
-        // ordered from first to last
-        return this.pastattempts.map(attempt => {
-          if (attempt.hinted) {
-            return 'hint';
+
+        // Create react component with current item data.
+        // If the component already existed, this will perform an update.
+        this.itemRenderer =
+        this.reactDOM.render( // eslint-disable-line new-cap
```
Comment (indirectlylit) [inline kolibri_exercise_perseus_plugin/assets/src/vue/index.vue:208]: whitespace issue in here

**Review by indirectlylit — APPROVED** [review]
looks good! notes:

* haven't been able to fully test
* could use a little cleanup of code formatting

**Comment by indirectlylit** [comment]
created an issue for outstanding cleanup. merging

**Pull request merged**